ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpConnection.java

```diff
@@ -0,0 +1,23 @@
+package io.ripc.protocol.tcp;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * An abstraction for a TCP connection.
+ *
+ * @param <R> The type of objects read from this connection.
+ * @param <W> The type of objects written to this connection.
+ */
+public interface TcpConnection<R, W> extends Publisher<R> {
```

Comment:
Although I'm not saying specific protocol don't have their modules, but we should have a common notion living in ripc core, named Connection or Channel or Pipe or whatever.
If we find anything specific to TCP in addition to write and subscribe, e.g. UDP or 0MQ specific helpers they will certainly be able to extend this common artifact. It will bring much more value to the core and a bunch of candidates protocol won't even bother going that far in their first iteration. RIPC is about connecting and moving data between processes so we must make sure the core interface represents that. Server and Client can also do the same, and you already have this notion in RxNetty (as in reactor-net), another qualifier as generic IMO is Recipient and Sender. This inherent to any communication.